src/textEditor.ts

```diff
@@ -140,6 +141,32 @@ export class TextEditor {
     return vscode.window.activeTextEditor.document.getText(selection);
   }
 
+  /**
+   *  Retrieves the current word at position.
+   *  If current position is whitespace, selects the right-closest word
+   */
+  static getWord(position: Position) : string {
+    const whitespaceRegex = new RegExp("^ *$");
+    let start = position;
+    let end = position.getRight();
+
+    const char = TextEditor.getText(new vscode.Range(start, end));
+    if (whitespaceRegex.test(char)) {
+      start = position.getWordRight();
+    } else {
+      start = position.getWordLeft(true);
+    }
+    end = start.getCurrentWordEnd(true).getRight();
+
+    const word = TextEditor.getText(new vscode.Range(start, end));
+
+    if (whitespaceRegex.test(word)) {
+      throw VimError.fromCode(ErrorCode.E348);
```

Comment:
> As it is, anyone else who calls this function will have to remember that it throws or else they'll get into trouble.

Fair point. Coming from c#, the compiler would complain if you didn't properly catch an exception.